mbuild/compound.py

```diff
@@ -2516,11 +2519,28 @@ def _energy_minimize_openbabel(
                         )  # openbabel indices start at 1
                         ob_constraints.AddIgnore(pid)
 
-        obConversion = openbabel.OBConversion()
-        obConversion.SetInAndOutFormats("mol2", "pdb")
         mol = openbabel.OBMol()
 
-        obConversion.ReadFile(mol, os.path.join(tmp_dir, "un-minimized.mol2"))
+        # convert compound to openbabel mol
+        ids = {}
+        for i, part in enumerate(self):
+            ids[id(part)] = i + 1
+            a = mol.NewAtom()
+            a.SetVector(
+                np.round(part.pos[0] * 10.0, 4),
+                np.round(part.pos[1] * 10.0, 4),
+                np.round(part.pos[2] * 10.0, 4),
+            )
+            a.SetAtomicNum(part.element.atomic_number)
+            a.SetType(part.element.symbol)
+
+        # since mbuild doesn't consider bond order,
+        # we will set bond order to 1 and call PerceiveBondOrders
+        for bond in self.bonds():
+            mol.AddBond(ids[id(bond[0])], ids[id(bond[1])], 1)
+
+        mol.PerceiveBondOrders()
+        mol.SetAtomTypesPerceived()
```

Comment:
I think we actually have a method doing this already in `conversion.py` (called `to_pybel`, also can be called with `Compound.to_pybel()`). It miss the `PerceiveBondOrders()` and `SetAtomTypesPerceived()` I believe, but I think you can use the other as base and just call the extra function.

Reply:
I did this way initially because I assumed it would be faster than using to_pybel.  It does not appear to be any different <shrug>, so I guess no point in using a different function.

I'll note that the "mol" returned by `to_pybel` is of type `openbabel.pybel.OBMol` (since it is using the pybel wrapper for openbabel).  In the code above, I'm calling `mol = openbabel.OBMol()` which is type `openbabel.openbabel.OBMol` (i.e., the standard swig bindings to openbabel).   Pybel doesn't have as much wrapped, but you can just access the swig bindings by inserting an OBMol in the call (e.g.,` mol.OBMol.PerceiveBondOrders()` ).  Since everything was set up to work with and `openbabel.openbabel.OBMol`, the code I'm going to update will just do the following, so I don't have to change any other syntax. 

`mol = self.to_pybel()
mol = mol. OBMol`  

The `PerceiveBondOrders()` and `SetAtomTypesPerceived()` are likely unnecessary, but I'm pretty sure they were both called during the execution of reading a mol2 file in (i.e., the original way we had the code set up, which worked), so I added those in here, but I don't think they are necessary to add into the to_pybel.